docs/glossary.md

```diff
@@ -0,0 +1,258 @@
+verification status.
+
+**Data Verification** - The process of cryptographically verifying that
+retrieved data matches its data root hash, ensuring data integrity.
+
+**Observer** - A gateway component that monitors and reports on the health and
+behavior of other gateways in the AR.IO network.
+
+**X-AR-IO Headers** - HTTP headers added by the gateway providing metadata about
+data verification, caching, and trust status.
+
+**X-ArNS Headers** - HTTP response headers added during ArNS name resolution
+that provide metadata about the resolution process, including the resolved
+transaction ID, TTL, process ID, and undername information.
+
+## Databases
+
+**Bundles Database** - SQLite database storing ANS-104 bundle metadata, data
+items, and their relationships. Includes retry tracking for bundle processing.
+
+**Core Database** - Primary SQLite database containing blocks, transactions,
+transaction tags, stable/new data indexes, and the migrations table that tracks
+applied database schema changes.
+
+**Data Database** - SQLite database tracking contiguous data availability,
+verification status, and retry attempts.
+
+**Moderation Database** - SQLite database managing content blocking and
+filtering rules.
+
+## Data Storage Architecture
+
+**Cache Store** - High-speed storage layer (Redis or filesystem) for frequently
+accessed data chunks and headers.
+
+**Chunk Data Store** - Backend storage for transaction chunks. Supports multiple
+implementations including filesystem and S3.
+
+**Contiguous Data Store** - Storage backend for complete transaction data.
+Manages both data files and verification metadata.
+
+**Verification Priority** - Numeric value determining the order of data
+verification. Higher priority items are verified first.
+
+**Verification Retry Count** - Number of failed verification attempts for a
+piece of data. Used with exponential backoff to manage retries.
+
+## Resolution
+
+**Name Resolution** - The process of converting names or identifiers to
+transaction IDs. The primary type is ArNS resolution, which maps human-readable
+names (like "my-app") to their corresponding Arweave transaction IDs.
+
+**Path Resolution** - The process of interpreting URL paths to determine which
+transaction data to serve. Includes manifest resolution (looking up paths in a
+manifest's routing table), index path resolution (adding index.html), and
+fallback path handling for 404 errors.
+
+**Manifest** - A special JSON document that maps paths to transaction IDs,
+enabling directory-like navigation of Arweave data.
+
+**Sandbox** - A security mechanism that redirects transaction access to unique
+subdomains based on the transaction ID. Each transaction gets its own
+base32-encoded subdomain, providing browser origin isolation between different
+applications and content.
+
+## Worker Architecture
+
+**ANS-104 Unbundler** - Worker process that extracts data items from bundle
+transactions and queues them for indexing.
```

Comment:
Remind me: are we always serving the unbundled data items from root parent offsets (when they're not brought in via other caches/sources)? I ask mainly to note that we could use qualifiers here to highlight the important of the indexing.